src/link.cpp

```diff
@@ -164,6 +164,34 @@ static void add_rpath(LinkJob *lj, Buf *rpath) {
     lj->rpath_table.put(rpath, true);
 }
 
+static Buf *get_dynamic_linker_path(CodeGen *g) {
+    if (g->is_native_target && g->zig_target.arch.arch == ZigLLVM_x86_64) {
+        const char *cc_exe = getenv("CC");
+        cc_exe = (cc_exe == nullptr) ? "cc" : cc_exe;
+        ZigList<const char *> args = {};
+        args.append("-print-file-name=ld-linux-x86-64.so.2");
+        Termination term;
+        Buf *out_stderr = buf_alloc();
+        Buf *out_stdout = buf_alloc();
+        int err;
+        if ((err = os_exec_process(cc_exe, args, &term, out_stderr, out_stdout))) {
+            return target_dynamic_linker(&g->zig_target);
+        }
+        if (term.how != TerminationIdClean || term.code != 0) {
+            return target_dynamic_linker(&g->zig_target);
+        }
+        if (buf_ends_with_str(out_stdout, "\n")) {
+            buf_resize(out_stdout, buf_len(out_stdout) - 1);
+        }
+        if (buf_len(out_stdout) == 0 || buf_eql_str(out_stdout, "ld-linux-x86-64.so.2")) {
+            return target_dynamic_linker(&g->zig_target);
+        }
+        return out_stdout;
```

Comment:
Can this ever return anything but `/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2`?

Reply:
on nixos you can have multiple shells open with different dynamic linker versions. here's an example value for me:
```
#define ZIG_DYNAMIC_LINKER "/nix/store/2kcrj1ksd2a14bm5sky182fv2xwfhfap-glibc-2.26-131/lib/ld-linux-x86-64.so.2"
```